src/main/java/core/basesyntax/OperationStrategyImpl.java

```diff
@@ -0,0 +1,15 @@
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(
+            Map<FruitTransaction.Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
+        return operationHandlers.get(operation);
+    }
+}
```

Comment:
It's recommended to place this class into the `strategy` package to better organize the code structure and adhere to the checklist item regarding not keeping all logic in a single package.